src/schemas/Clinical/index.ts

```diff
@@ -133,39 +133,129 @@ const convertClinicalSubmissionDataToGql = (
       updatedAt: string;
       clinicalEntities: { [k: string]: SubmissionEntity };
     };
-    batchErrors?: unknown[];
-  },
-) => {
-  const submission = get(data, 'submission', {} as Partial<typeof data.submission>);
-  const fileErrors = get(data, 'batchErrors', [] as typeof data.batchErrors);
-  const clinicalEntities = get(
-    submission,
-    'clinicalEntities',
-    {} as typeof data.submission.clinicalEntities,
-  );
+      batchErrors?: unknown[];
+    },
+  ) => {
+    const submission = get(data, 'submission', {} as Partial<typeof data.submission>);
+    const fileErrors = get(data, 'batchErrors', [] as typeof data.batchErrors);
+    const clinicalEntities = get(
+      submission,
+      'clinicalEntities',
+      {} as typeof data.submission.clinicalEntities,
+    );
+  
+    return {
+      id: submission._id || null,
+      programShortName,
+      state: submission.state || null,
+      version: submission.version || null,
+      updatedBy: submission.updatedBy || null,
+      updatedAt: submission.updatedAt ? new Date(submission.updatedAt) : null,
+      clinicalEntities: async () => {
+        const clinicalSubmissionTypeList = await clinicalService.getClinicalSubmissionTypesList();
+        const filledClinicalEntities = clinicalSubmissionTypeList.map(clinicalType => ({
+          clinicalType,
+          ...(clinicalEntities[clinicalType] || {}),
+        }));
+        return filledClinicalEntities.map(clinicalEntity =>
+          convertClinicalSubmissionEntityToGql(clinicalEntity.clinicalType, clinicalEntity),
+        );
+      },
+      fileErrors: fileErrors?.map(convertClinicalFileErrorToGql),
+    };
+  };
 
-  return {
-    id: submission._id || null,
+type ClinicalEntityData = {
+  programShortName: string;
+  clinicalEntities: ClinicalEntityRecord[];
+  completionStats: CompletionStats[];
+};
+
+type ClinicalVariables = {
+    programShortName: string;
+    filters: {
+      entityTypes: string[];
+      page: number;
+      limit: number; 
+      donorIds: number[];
+      submitterDonorIds: string[];
+      completionState: string;
+      sort: string;
+    }, 
+}
+
+enum CoreClinicalEntities {
+  'donor',
+  'specimens',
+  'primaryDiagnosis',
+  'followUps',
+  'treatments',
+  'familyHistory'
+}
```

Comment:
There is nothing wrong with this, necessarily, but you might want to define each entry with a string value:
```
enum CoreClinicalEntities {
  donor='donor',
  specimens='specimens',
  primaryDiagnosis='primaryDiagnosis',
  followUps='followUps',
  treatments='treatments',
  familyHistory='familyHistory'
}
```
As written, they are assigned numbers based on their order in this list (CoreClinicalEntities.donor = 0, etc.). This is relevant when you create an object of a type like:
```
type CoreCompletionFields = {
  [k in CoreClinicalEntities]?: number;
};
```
Any object of this type will looke like:
```
{ 0: {...donorData}, 1: {...specimenData} }
```